test/test_queue.rb

```diff
@@ -24,4 +24,30 @@ def test_flaky_jobs
 
     assert_failures(["./spec/legit_failure_spec.rb[1:3]"], queue)
   end
+
+  def test_fail_fast
+    build_id = rand_id
+
+    Process.wait(start_worker(
+      build_id: build_id, suite: "failing_suite"))
+
+    queue = RSpecQ::Queue.new(build_id, "foo", REDIS_HOST, 1)
+
+    assert_queue_well_formed(queue)
+
+    assert queue.suite_failed?
+  end
+
+  def test_no_fail_fast
+    build_id = rand_id
+
+    Process.wait(start_worker(
+      build_id: build_id, suite: "failing_suite"))
+
+    queue = RSpecQ::Queue.new(build_id, "foo", REDIS_HOST)
+
+    assert_queue_well_formed(queue)
+
+    refute queue.suite_failed?
+  end
```

Comment:
This new functionality should also be covered by an end-to-end test. See `test/test_e2e.rb` for other examples.